### ISSUE TITLE
Add support for MySQL's BIN_TO_UUID and UUID_TO_BIN functions

### ISSUE THREAD

**Issue by sivaraam** [jOOQ/jOOQ#16524]
### Expected behavior

I am successfully able to parse a query that uses MySQL's `BIN_TO_UUID` / `UUID_TO_BIN` function.

Reference link to MySQL doc that mentions the functions: https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html

### Actual behavior

Parsing a query with MySQL's `BIN_TO_UUID` / `UUID_TO_BIN` function seems to fail.

### Steps to reproduce the problem

Parse the following SQL using jOOQ:
```sql
select BIN_TO_UUID(id), a, b from example e
```

Code used to parse the query

```java
String sql = "select BIN_TO_UUID(id), a, b from example e";
SelectQuery<Record> sel = (SelectQuery<Record>) dslContext.parser().parseSelect(sql);
```

### jOOQ Version

3.19.6

### Database product and version

MySQL 8

### JDBC / R2DBC driver name and version (include name if unofficial driver)

com.mysql.jdbc.Driver

### Dialect support

- [x] ClickHouse: https://clickhouse.com/docs/en/sql-reference/functions/uuid-functions#uuidstringtonum
- [x] CockroachDB: https://www.cockroachlabs.com/docs/stable/uuid#supported-casting-and-conversion
- [x] H2: `CAST(bytes AS UUID)` and `CAST(uuid AS BINARY(16))`
- [x] HSQLDB: `CAST(bytes AS UUID)` and `CAST(uuid AS BINARY(16))`
- [x] MySQL: https://dev.mysql.com/doc/refman/8.3/en/miscellaneous-functions.html
- [x] PostgreSQL: `decode(replace(cast(uuid as text), '-', ''), 'hex')` and `cast(encode(bytes, 'hex') as uuid)`
- [x] SQL Server: `CAST(bytes AS UNIQUEIDENTIFIER)` and `CAST(uuid AS BINARY(16))`
- [x] YugabyteDB: Like PostgreSQL

**Comment by lukaseder** [comment]
Thanks for your feature request. Related:

- https://github.com/jOOQ/jOOQ/issues/11807

It will make sense to support binary / uuid conversion functions across dialects, as `binary(16)` is a common representation of a `UUID` value, if native support is not available.

In the meantime, depending on your parser use-cases, you can either:

- Use `Settings.parseUnknownFunctions` to ignore such functions: https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/custom-settings/settings-parser/
- Use a `ParseListener` to extend the parser to support these functions: https://www.jooq.org/doc/latest/manual/sql-building/sql-parser/sql-parser-listener/

**Comment by sivaraam** [comment]
> Thanks for your feature request. Related:
> 

Thank you for the prompt response!

>   * [Add a new SQLDataType.BINARY_UUID type #11807](https://github.com/jOOQ/jOOQ/issues/11807)
>
> It will make sense to support binary / uuid conversion functions across dialects, as `binary(16)` is a common representation of a `UUID` value, if native support is not available.

Indeed. Those seems like a good enhancement for jOOQ 🙂 
 
> 
> In the meantime, depending on your parser use-cases, you can either:
> 
>   * Use `Settings.parseUnknownFunctions` to ignore such functions: https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/custom-settings/settings-parser/
>   * Use a `ParseListener` to extend the parser to support these functions: https://www.jooq.org/doc/latest/manual/sql-building/sql-parser/sql-parser-listener/

Thanks for the helpful pointers. Will definitely look into them 👍🏼

**Comment by lukaseder** [comment]
The emulation will work the other way round as well, i.e. some dialects support `CAST(<binary> AS UUID)` and `CAST(<uuid> AS BINARY)`, including CockroachDB, H2, HSQLDB, MariaDB:

- https://github.com/jOOQ/jOOQ/issues/16526

**Comment by lukaseder** [comment]
More emulations will be done via a separate task:

- https://github.com/jOOQ/jOOQ/issues/16527

**Comment by lukaseder** [comment]
Implemented in 3.20.

**Comment by sivaraam** [comment]
> Implemented in 3.20.

Great to hear that! Thanks for the quick fix 👍🏼
